File: api/Entities/Admin/Order/Order.cs

using System.ComponentModel.DataAnnotations;
using api.Entities.Admin.Client;

namespace api.Entities.Admin.Order
{
    public class Order: BaseEntity
    {
        [MaxLength(1), Required, RegularExpression(@"^[DdIi]*$", ErrorMessage ="Destination value should be either d or D (for domestic recruitment), or I or i (for international recruitments)")]
        public string Destination {get; set;}   //domestic or international
        public int OrderNo { get; set; }
        public DateTime OrderDate { get; set; }=DateTime.UtcNow;
        public int CustomerId { get; set; }
        public string OrderRef { get; set; }
        public DateTime OrderRefDate {get; set;}
        public int ProjectManagerId { get; set; }
        public int? SalesmanId { get; set; }
        
        public DateTime CompleteBy { get; set; }
        public string Country { get; set; }
        public string CityOfWorking { get; set; }
        public int ContractReviewId { get; set; }
        public string ContractReviewStatus {get; set;}="Not Reviewed";
        public string Status { get; set; } = "Awaiting Review";
        public DateTime? ForwardedToHRDeptOn { get; set; }
        public ICollection<OrderItem> OrderItems { get; set; }
        public Customer Customer {get; set;}
        public ContractReview ContractReview {get; set;}
        public ICollection<OrderForwardToAgent> DLForwarded {get; set;}
        
    }
}

File: client/src/app/_models/admin/order.ts
import { IOrderItem } from "./orderItem";

export interface IOrder 
{
     id: number; 
     orderNo: number; 
     orderDate: Date; 
     customerId: number; 
     customerName: string;
     buyerEmail: string; 
     orderRef: string; 
     orderRefDate: Date;
     salesmanId: number; 
     salesmanName: string; 
     projectManagerId: number;
     //medicalProcessInchargeEmpId: number; 
     //visaProcessInchargeEmpId: number; 
     //emigProcessInchargeId: number;
     //travelProcessInchargeId: number;
     completeBy: Date; 
     country: string; 
     cityOfWorking: string; 
     contractReviewStatus: string;
     estimatedRevenue: number;
     status: string; 
     forwardedToHRDeptOn: Date; 
     orderItems: IOrderItem[];
     
}

export class Order implements IOrder
{
     id=0; 
     orderNo= 0; 
     orderDate = new Date(); 
     customerId= 0; 
     customerName= '';
     buyerEmail= ''; 
     orderRef= ''; 
     orderRefDate= new Date;
     salesmanId= 0; 
     salesmanName= ''; 
     projectManagerId= 0;
     medicalProcessInchargeEmpId= 0; 
     visaProcessInchargeEmpId= 0; 
     emigProcessInchargeId= 0;
     travelProcessInchargeId= 0;
     completeBy= new Date; 
     country= ''; 
     cityOfWorking= ''; 
     contractReviewStatus = ''
     estimatedRevenue= 0;
     status= ''; 
     forwardedToHRDeptOn= new Date('1900-01-01'); 
     orderItems: IOrderItem[]=[];
     
}